### PR TITLE
Draft Mechanics script fix #137

### LINKED ISSUE #137: Script:"Mechanics of Contribution"
We should have a detailed-enough script for the "mechanics of contribution" segment of the Contributor section.

### PR DESCRIPTION
**Pull request by lenucksi**
Here's a draft PR for the mechanics of contributing script.
I'll add more text later from the articles, this is for now just the recording of meeting notes with @MaineC and @lenucksi.

### CODE REVIEW

**Comment by rrrutledge** [comment]
👍.  Thanks for sharing.

**Comment by rrrutledge** [comment]
This addresses https://github.com/InnerSourceCommons/InnerSourceLearningPath/issues/160

**Comment by lenucksi** [comment]
I'll add more content on the weekend, however likely mostly appending to the existing content.
The existing content can be given a first review, later reviews against additional content are very welcome too.

Keep in mind that this is slightly more WIP-y content that the usual pull requests, though.

File: contributor/04-mechanics-of-contributing-script.md [hunk 1]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
 
 ## Script
 
+X: In the previous video we learned about the contributor ethos and the guest-in-home analogy.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
I: In the previous video we learned about the contributor ethos and the guest-in-home analogy. 
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 2]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+
+X: Try to make time to meet the people behind the names. If possible, try to meet them over your favorite beverage or some food. 
+If that's not possible having a relaxed informal water-cooler style chat about whatever via your video conference system can provide some of this bandwidth.
+When you're able to hear people speak, when you know their idiosyncrasies remote collaboration will become easier. 
+
+X: The contributions you're going to make will vary in size and scope - sometimes it's just a bug fix, documentation amendment, other times it will be larger feature.
+A time-and-again-proven wisdom from the Open Source arena, especially for large contributions, is to get into contact with a Trusted Committer early to get agreement on the user/technical design of the contribution _before_ working on the changes and submitting a pull request.
+Also, working in the open as early as possible can help you from going down a wrong path, loosing time, wasting effort or even being rejected.
+
+X: A good Trusted Committer will help and mentor you on your way and not expect only completely finished, perfect ready-to-nod-off reviewable content.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
J: A good Trusted Committer will help and mentor you on your way and not expect only completely finished, perfect ready-to-nod-through reviewable content.   
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 3]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+
+X: The contributions you're going to make will vary in size and scope - sometimes it's just a bug fix, documentation amendment, other times it will be larger feature.
+A time-and-again-proven wisdom from the Open Source arena, especially for large contributions, is to get into contact with a Trusted Committer early to get agreement on the user/technical design of the contribution _before_ working on the changes and submitting a pull request.
+Also, working in the open as early as possible can help you from going down a wrong path, loosing time, wasting effort or even being rejected.
+
+X: A good Trusted Committer will help and mentor you on your way and not expect only completely finished, perfect ready-to-nod-off reviewable content.   
+
+-> Show phases image, crafting part
+
+X: Let's talk about the gotchas awaiting you now that you're familiar with the host team's approach and them looking forward to your pull request.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
J: Let's talk about the gotchas awaiting you now that you're familiar with the host team's approach and them looking forward to your pull request. 
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 4]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+
+X: A good Trusted Committer will help and mentor you on your way and not expect only completely finished, perfect ready-to-nod-off reviewable content.   
+
+-> Show phases image, crafting part
+
+X: Let's talk about the gotchas awaiting you now that you're familiar with the host team's approach and them looking forward to your pull request. 
+First, you'll be in less direct contact with them and second you aren't expected to be as knowledgeable and proficient as you might be on the full-time projects that your team owns.
+How can you now deal with this the best?
+
+X: Being InnerSource and thus written-communication focused their documentation, the conversation archives and code artifacts from the host team should serve well as a way to unblock yourself.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
I: Being InnerSource and thus written-communication focused their documentation, the conversation archives and code artifacts from the host team should serve you well as a way to unblock yourself.
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 5]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+This is similar to the situation you and likely most people find yourself in when using one of the popular OSS projects, such as [I'd put OpenZFS, GoCD or OSM here, we'll need another, popular/well-known, well documented more library-style project with a great, helpful community].
+
+X: Do you find that things are going nowhere even after trying to unblock yourself?
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
J: Do you find that things are going nowhere even after trying to unblock yourself?
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 6]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+X: A good Trusted Committer will help and mentor you on your way and not expect only completely finished, perfect ready-to-nod-off reviewable content.   
+
+-> Show phases image, crafting part
+
+X: Let's talk about the gotchas awaiting you now that you're familiar with the host team's approach and them looking forward to your pull request. 
+First, you'll be in less direct contact with them and second you aren't expected to be as knowledgeable and proficient as you might be on the full-time projects that your team owns.
+How can you now deal with this the best?
+
+X: Being InnerSource and thus written-communication focused their documentation, the conversation archives and code artifacts from the host team should serve well as a way to unblock yourself.
+This is similar to the situation you and likely most people find yourself in when using one of the popular OSS projects, such as [I'd put OpenZFS, GoCD or OSM here, we'll need another, popular/well-known, well documented more library-style project with a great, helpful community].
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
This is similar to the situation you and likely most people find yourself in when using one of the popular OSS projects, such as [I'd put OpenZFS, GoCD or OSM, maybe Apache CouchDB, Apache Beam here, we'll need another, popular/well-known, well documented more library-style project with a great, helpful community].
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 7]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+
+X: Do you find that things are going nowhere even after trying to unblock yourself?
+Try to get in contact with the project via one of it's trusted comitters. Hey, they may even be as close as one floor away, why not just drop by for a coffee?
+While that might solve *your* problem quickly, it will require the TC's availability and time and leave the problem open for the next person having the same one.
+
+X: In most cases InnerSource projects will have public, text-based, searchable and archived communication groups, such as a mailing list, a chat room, the issue tracker or something similar depending on the purpose of having a more synchronous or asynchronous way of interacting.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
J: In most cases InnerSource projects will have public, text-based, searchable and archived communication groups, such as a mailing list, a chat room, the issue tracker or something similar depending on the purpose of having a more synchronous or asynchronous way of interacting.
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 8]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+Using them will bring your problem in front of more people who might be able to help you and, more importantly, will record your problem and its possible solution to more people having the same problem in the future.
+
+X: Think about how you might utilize StackOverflow or GitHub bug trackers in your daily work. We can possibly all remember a solution to some odd but critical quirk in one of the libraries our software uses that was caught by happenstance on some bug tracker.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
J: Think about how you might utilize StackOverflow or GitHub bug trackers in your daily work. We can possibly all remember a solution to some odd but critical quirk in one of the libraries our software uses that was caught by happenstance on some bug tracker. 
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 9]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+
+X: In most cases InnerSource projects will have public, text-based, searchable and archived communication groups, such as a mailing list, a chat room, the issue tracker or something similar depending on the purpose of having a more synchronous or asynchronous way of interacting.
+Using them will bring your problem in front of more people who might be able to help you and, more importantly, will record your problem and its possible solution to more people having the same problem in the future.
+
+X: Think about how you might utilize StackOverflow or GitHub bug trackers in your daily work. We can possibly all remember a solution to some odd but critical quirk in one of the libraries our software uses that was caught by happenstance on some bug tracker. 
+Documentation built up en passant is called passive documentation and this approach can be of great help in the InnerSource context too.
+
+X: Did you figure out an error in a piece of existing documentation in the host team's project? Please consider updating it, making a quick pull request or reaching out to the project if there's no easy way to provide the update yourself.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
I: Did you figure out an error in a piece of existing documentation in the host team's project? Please consider updating it, making a quick pull request or reaching out to the project if there's no easy way to provide the update yourself.
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 10]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+Hosts teams or future users and contributors will probably be very happy. As a side-effect this might already establish a bit of trust with the host project making it easier for you to get contributions merged in the future. 
+
+-> Show phases image, polishing & wrapping part
+
+X: So you've crafted a nice addition to the hosts teams project, added good documentation pieces and circumvented all the small quirks and large dragons looming on the way with the help of the community and maybe even fixed a small doc mistake on the way. How do you make sure your contribution gets reviewed and merged as quick and easy as possible?
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
J: So you've crafted a nice addition to the hosts teams project, added good documentation pieces and circumvented all the small quirks and large dragons looming on the way with the help of the community and maybe even fixed a small doc mistake on the way. How do you make sure your contribution gets reviewed and merged as quick and easy as possible?
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 11]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+
+X: Most of this is about enabling a trusted committer to review your contribution asynchronously without your presence as well as enabling them to focus on questions that would not be solvable in an automated fashion, such as questions of code style, evaluating possibly regressions, etc.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
I: Most of this is about enabling a trusted committer to review your contribution asynchronously without your presence as well as enabling them to focus on questions that would not be solvable in an automated fashion, such as questions of code style, evaluating possibly regressions, etc.
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 12]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+
+[Might be nice to be able to reference this: http://commadot.com/wtf-per-minute/ ]
+
+X: A great starting point to find out what a host teams likes with regard to their code style, testing setup and more on their development workflow is a document you might have taken a look at while initially setting up your workspace -their CONTRIBUTING.md and sometimes their README.md files. Definitely take a look at them, every good InnerSource project should have them.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
J: A great starting point to find out what a host teams likes with regard to their code style, testing setup and more on their development workflow is a document you might have taken a look at while initially setting up your workspace -their CONTRIBUTING.md and sometimes their README.md files. Definitely take a look at them, every good InnerSource project should have them.
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 13]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+Even if you might not 100% agree with their code style decisions or other things - it's their house, please adhere to their wishes for now.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
Even if you might not agree with their code style decisions or other things 100% - it's their house, please adhere to their wishes for now.
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 14]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+
+X: Should you find that the CONTRIBUTING.md file does not answer your question definitely reach out to the host project as mentioned above.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
I: Should you find that the CONTRIBUTING.md file does not answer your question definitely reach out to the host project as mentioned above.
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 15]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+
+-> show link to innersourcecommons for link to contributing.md template?
+
+X: Often projects will provide scripts to check for compliance with their coding standards, passing tests, etc. before committing things - if your host projects provide such pre-submit scripts, definitely use them, they will save you and you the trusted comitter time.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
J: Often projects will provide scripts to check for compliance with their coding standards, passing tests, etc. before committing things - if your host projects provide such pre-submit scripts, definitely use them, they will save you and the trusted comitters time.
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 16]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+
+X: Meaningful tests for your contribution are important with InnerSource as they will facilitate asynchronous validation of the work you crafted and allow everyone to check for regressions early on.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
J: Meaningful tests for your contribution are important with InnerSource as they will facilitate asynchronous validation of the work you crafted and allow everyone to check for regressions early on.
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 17]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+Often, InnerSource projects will have larger automated integration tests and CI suites running to enable validation of pull requests and the current trunk in a larger context.
+Don't forget to regularly validate that what you've built solves your own problem that made you consider contributing in the first place! 
+
+X: The articles accompanying this video have more technical details on testing and validation, feel welcome to give them a read. Let's come a to a last aspect: Try to be comfortable working in the open.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
I: The articles accompanying this video have more technical details on testing and validation, feel welcome to give them a read. Let's come to a last aspect: Try to be comfortable working in the open. 
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 18]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+
+X: Meaningful tests for your contribution are important with InnerSource as they will facilitate asynchronous validation of the work you crafted and allow everyone to check for regressions early on.
+Often, InnerSource projects will have larger automated integration tests and CI suites running to enable validation of pull requests and the current trunk in a larger context.
+Don't forget to regularly validate that what you've built solves your own problem that made you consider contributing in the first place! 
+
+X: The articles accompanying this video have more technical details on testing and validation, feel welcome to give them a read. Let's come a to a last aspect: Try to be comfortable working in the open. 
+
+X: Just open up a branch, pull request it, mark it as WIP and start crafting code once you've reached a rough agreement on how to bring up your contribution as we discussed earlier. You'll be able to get early feedback from a trusted committer or others preventing you from running in the wrong direction or walking down dead alleys.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
J: Just open up a branch, pull request it, mark it as WIP and start crafting code once you've reached a rough agreement on how to bring up your contribution as we discussed earlier. You'll be able to get early feedback from a trusted committer or others preventing you from running in the wrong direction.
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 19]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+A good trusted comitter should not look for just "nod off" pull requests but be happy to mentor and help you along the way.
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
A good trusted committer will not look for "nod through" pull requests but be happy to mentor and help you along the way.
```

File: contributor/04-mechanics-of-contributing-script.md [hunk 20]
```diff
@@ -10,3 +10,159 @@ What are the nuts and bolts of advice to follow when acting as an InnerSource Co
...
+
+X: Finally, remember that nobody is perfect, should be perfect and that mastery comes through a cycle of trying, failing, learning and trying again. Give it your best shot and have fun!
```
Comment (MaineC) [inline contributor/04-mechanics-of-contributing-script.md]:
```suggestion
J: Finally, remember that nobody is perfect, should be perfect and that mastery comes through a cycle of trying, failing, learning and trying again. Give it your best shot and have fun! 
```

**Comment by lenucksi** [comment]
Please open a second PR with changes if required.

**Pull request merged**